mllib/src/test/scala/org/apache/spark/ml/util/DefaultReadWriteTest.scala

```diff
@@ -107,24 +133,34 @@ trait DefaultReadWriteTest extends TempDirectory { self: Suite =>
     }
     val model = estimator.fit(dataset)
 
-    // Test Estimator save/load
-    val estimator2 = testDefaultReadWrite(estimator)
-    testParams.foreach { case (p, v) =>
-      val param = estimator.getParam(p)
-      assert(estimator.get(param).get === estimator2.get(param).get)
+    val testFunctions = if (testParams.contains("initialModel")) {
+      Map(("initialModel", checkModelData.asInstanceOf[(Any, Any) => Unit]))
```

Comment:
One possible solution on this is overriding the equal method for the base Model class so that we needn't override the equal method to all Models.